모임 단건 조회 기능 구현

hyeonnkang opened the pull request:
## 관련 이슈
- close #62 

## 작업 사항
- erd 커밋을 docs로 변경
- Meet Builder 에서 createdBy 및 updatedBy 설정 삭제
- Meet의 toString 삭제
- MeetController의 api url meets 로 변경
- MeetServiceTest 삭제 ( 이유 : Meet 생성 시 Member 넣어 주기 에서 오류가 발생해 이를 해결하지 못했습니다. swagger 테스트는 잘 작동하는 상태입니다.)

## 참고 자료
- 참고자료 (링크 등)

## 체크리스트
- [x]  PR의 제목은 팀 내 규칙을 준수하여 알맞게 작성하였는가?
- [x]  Merge 하는 브랜치가 올바른가? (`main` branch에 실수로 PR 생성 금지)
- [x]  팀의 코딩 컨벤션을 준수하는가?
- [x]  PR과 관련없는 변경사항이 들어가지는 않았는가?
- [x]  내 코드에 대한 자기 검토가 되었는가?
- [x]  Reviewers, Assignees, Lables, Project, Milestone은 적절하게 선택하였는가?
- [x]  관련한 issue를 닫아야 하는지 점검해보고 적용했는가?

hyeonnkang committed:
docs: #62 erd 수정 (Meet 테이블에서 leader_id 제거)

Meet 테이블에서 created_by와 중복되는 내용이라 leader_id를 제거하고 해당 erd로 변경해주었다.

hyeonnkang committed:
feat: #62 모임 단건 조회 기능 적용 및 swagger 테스트

모임 단건 조회 적용 및 swagger 테스트 완료했습니다.
- createdBy updatedBy 삭제
- to String 삭제
- meets 로 api url 변경

MeetServiceTest는 Meet 생성 시 Member 설정 오류가 나는 것 같은데, MemberServiceTest 코드를 참고해도 이해가 잘 가지 않아 현재 아예 삭제했습니다!

Wo-ogie commented:
Issue 그냥 #45 reopen해서 하시면 되지 않을까요? 굳이 issue 따로 만들어서 안해도 될 것 같아요.
+) 엇 근데 이미 commit에도 이슈 번호를 붙이셔서 다시 하기 좀 귀찮겠네요. 그냥 이번에는 이렇게 하시고, 다음에는 기존 issue를 reopen하는 방법도 고려하시면 좋을 것 같습니다~

Wo-ogie commented on src/main/java/com/umc/withme/controller/MeetController.java:
```diff
@@ -27,12 +34,21 @@ public class MeetController {
+    })
+    @PostMapping("/meets")
     public ResponseEntity<DataResponse<MeetCreateResponse>> createMeet(
             @Valid @RequestBody MeetCreateRequest meetCreateRequest,
-            @RequestParam Long memberId // TODO : 인증기능 구현되면 로그인 사용자 정보 가져오는 걸로 변경
+            @Parameter(hidden = true) @AuthenticationPrincipal WithMeAppPrinciple principle
     ) {
```
`MemberCreateResponse`, `MeetCreateRequest`에도 API 명세 작성해주세요. `MemberCreateResponse`와 `MeetCreateRequest` 안에 있는 DTO가 있다면 거기도 함께 작성해주시면 좋을 것 같습니다.
`@Schema`로 example 값이랑 description 작성할 수 있고, 제가 한 거 참고하시면 됩니다.

Wo-ogie commented on src/main/java/com/umc/withme/controller/MeetController.java:
```diff
@@ -41,4 +57,32 @@ public ResponseEntity<DataResponse<MeetCreateResponse>> createMeet(
+            @ApiResponse(responseCode = "404", description = "1401: <code>email</code>에 해당하는 회원이 없는 경우", content = @Content)
+    })
+    @GetMapping("/meets/{meetId}")
+    public ResponseEntity<DataResponse<MeetInfoGetResponse>> getMeet(@PathVariable Long meetId) {
```
여기도 `MeetInfoGetResponse`에 API 명세 작성해주세요. `MeetInfoGetResponse` 안에 있는 DTO가 있다면 거기도 함께 작성해주시면 좋을 것 같습니다.

Wo-ogie commented on src/main/java/com/umc/withme/service/MeetService.java:
```diff
@@ -23,18 +26,18 @@ public class MeetService {
-    public Long createMeet(MeetDto meetDto, Long leaderId) {
-        Member leader = memberRepository.findById(leaderId)
-                .orElseThrow(EntityNotFoundException::new); // TODO : 추후 인증기능 구현되면 exception 교체해야된다.
+    public Long createMeet(MeetDto meetDto, String leaderEmail) {
+        Member leader = memberRepository.findByEmail(leaderEmail)
+                .orElseThrow(UnauthorizedException::new);
```
로그인 사용자가 없는 경우에 대한 예외처리는 따로 하고 있으니, 그냥 `EmailNotFoundException`을 throw하셔도 될 것 같아요.

Wo-ogie commented on src/main/java/com/umc/withme/controller/MeetController.java:
```diff
@@ -41,4 +57,32 @@ public ResponseEntity<DataResponse<MeetCreateResponse>> createMeet(
+     * @param meetId 조회하려는 모임의 id
+     * @return 조회한 모임의 정보를 담은 MeetInfoGetResponse를 data에 담아서 반환한다.
+     */
+    @Operation(
+            summary = "모임 모집글 1개 조회 API",
+            description = "<p><code>meetId</code>에 해당하는 모임의의 정보를 request body 에 넣어 전달합니다.</p>",
+            security = @SecurityRequirement(name = "access-token")
+    )
+    @ApiResponses({
+            @ApiResponse(responseCode = "200", description = "Success"),
+            @ApiResponse(responseCode = "404", description = "1401: <code>email</code>에 해당하는 회원이 없는 경우", content = @Content)
+    })
```
여기 `@Operation` description 부분에 오타 있어요.

그리고 `@ApiResponse`는 해당 api에 대해 명시해야 할 사항이 경우에만 작성해주시면 될 것 같습니다. 모든 예외 상황에 대해서 다 하려면 너무 코드가 길어질 것 같아서요. 그래서 404는 굳이 없어도 되지 않을까..? 싶어요. 어떻게 생각하실까요?
이렇게 말하고 제 코드를 보니 저도 굳이 작성 안했어도 됐을 것 같네요.

hyeonnkang replied:
넵 위에 말씀하신 피드백 모두 반영해서 푸쉬했습니다! 감사합니다!

hyeonnkang committed:
feat: #62 코드리뷰 반영

- MeetCreateRequest, MeetCreateResponse, MeetInfoGetResponse 에 @Schema로 API 명세 작성
- 로그인 사용자가 없는 경우의 예외처리는 따로 해주고 있으므로 
 MeetService의 createMeet 메소드에서 UnautorizedExample을 EmailNotFoundException 으로 변경
- MeetController 에서 오타 수정 및 불필요한 @ApiResponse 제거

Wo-ogie commented on src/main/java/com/umc/withme/dto/meet/MeetCreateRequest.java:
```diff
@@ -24,17 +18,37 @@
 @NoArgsConstructor(access = AccessLevel.PROTECTED)
 public class MeetCreateRequest {
 
+    @Schema(
+            example = "{" +
+                    "{sido='서울특별시', sgg='강남구'}, " +
+                    "{sido='서울특별시', sgg='노원구'}" +
+                    "}",
+            description = "모임에 설정된 주소(동네) 목록")
     @NotEmpty
     private List<AddressRequest> addresses;
```
이거 이렇게 하지 말고 그냥 `AddressRequest`에 `@Schema`로 명세 작성해주시면 됩니다.

Wo-ogie commented on src/main/java/com/umc/withme/dto/member/MemberInfoGetResponse.java:
```diff
@@ -10,10 +11,21 @@
+    @Schema(example = "010-1234-5678", description = "사용자의 휴대폰 번호")
     private String phoneNumber;
+
+    @Schema(example = "AGE_20_29", description = "사용자의 연령대")
     private Integer ageRange;
```
이거 `Integer` 값이라서 `20`, `30` 이렇게 응답되지 않나요?

Wo-ogie commented on src/main/java/com/umc/withme/dto/meet/MeetInfoGetResponse.java:
```diff
@@ -16,20 +17,51 @@
+                    "{sido='서울특별시', sgg='노원구'}" +
+                    "}",
+            description = "모임에 설정된 주소(동네) 목록")
     private List<AddressDto> addresses;  // 이 모임에 설정된 동네 리스트
```
여기도 `AddressDto`만 명세 작성해놓으면 그거 쓰도록 하는게 좋을 거 같아요.

Wo-ogie commented on src/main/java/com/umc/withme/dto/meet/MeetInfoGetResponse.java:
```diff
@@ -16,20 +17,51 @@
+    @Schema(example = "2", description = "조회한 모임의 id")
     private Long meetId;
+
+    @Schema(example = "", description = "모임을 생성한 사용자 정보 (모임의 리더 정보)")
     private MemberInfoGetResponse leader;  // 이 모임의 리더
```
`MemberInfoGetResponse` 명세가 표시되는 부분이라 `@Schema` 없어도 될 것 같습니다. 실제로 Swagger 돌려봐도 작성한 예시나 설명은 보이지 않네요.
굳이 사용하는 이유가 있다면 코드 상에서 명시할 수 있다는 정도인데, 이 부분은 굳이 필요하면 주석으로 사용하면 될 것 같습니다.
만약 `MemberInfoGetResponse` 전체에 대한 설명을 작성하고 싶으신거면 `MemberInfoGetResponse` class level에 `@Schema(description=)`을 붙이셔야 합니다. `example` 속성은 쓸 거 없으면 그냥 안쓰시면 되구요.

hyeonnkang replied:
넵 ageRange 변수명만 보고 example 작성했는데 Integer 값인거 확인하고 변경해놓았습니다. 나머지 부분들도 모두 수정해서 푸쉬했습니다!

hyeonnkang committed:
feat: #62 코드리뷰 반영

- API 명세 중복된 부분 삭제 후 DTO 내부에서 API 명세 작성
- ageRange Integer 값으로 받아오므로 example 적절하게 변경

Wo-ogie reviewed (approved):
수고하셨습니다! 모임 삭제 API PR 올리신 것도 여기에서 언급한 코멘트들에 해당 사항 없는지 한 번 확인해주시면 좋을 것 같아요.

Wo-ogie commented:
@sujeong11 혹시 시간 되시면 리뷰 부탁드립니다!

hyeonnkang committed:
Merge branch 'develop' into feature/#62-meet-findById
